src/nidigital/system_tests/test_system_nidigital.py

```diff
@@ -123,6 +119,69 @@ def get_test_file_path(test_name, file_name):
     return os.path.join(test_files_base_dir, test_name, file_name)
 
 
+@pytest.fixture(params=[array.array, numpy.array, list])
+def source_waveform_type(request):
+    if request.param != array.array:
+        pytest.skip('Source waveform types other than array.array are not supported yet (#1132)')
+    return request.param
+
+
+def test_source_waveform_parallel_site_unique(multi_instrument_session, source_waveform_type):
+    # Implementation of write_source_waveform_site_unique() uses
+    # memoryview of array.array, which is not supported in Python 2.x
+    if sys.version_info.major >= 3:
+        test_name = test_source_waveform_parallel_site_unique.__name__
+        configure_session(multi_instrument_session, test_name)
+
+        multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+
+        num_samples = 256
+        multi_instrument_session.write_sequencer_register(reg='reg0', value=num_samples)
+
+        multi_instrument_session.create_source_waveform_parallel(
+            pin_list='LowPins',
+            waveform_name='src_wfm',
+            data_mapping=2601)
```

Comment:
Is data mapping going to turn into an enum?

Reply:
Yes, it will done in #1066.